File: backend/src/models/ConnectionInvite.ts
import { randomBytes } from 'crypto';
import { HydratedDocument, Model, Schema, Types, model } from 'mongoose';
import { RelationshipType } from './Connection';

export interface ConnectionInvite {
  inviterId: Types.ObjectId;
  acceptedBy?: Types.ObjectId;
  token: string;
  relationshipType: RelationshipType;
  status: 'pending' | 'accepted' | 'expired' | 'cancelled';
  expiresAt: Date;
  createdAt: Date;
  updatedAt: Date;
}

const connectionInviteSchema = new Schema<ConnectionInvite>(
  {
    inviterId: {
      type: Schema.Types.ObjectId,
      ref: 'User',
      required: true,
      index: true,
    },
    acceptedBy: {
      type: Schema.Types.ObjectId,
      ref: 'User',
    },
    token: {
      type: String,
      required: true,
      unique: true,
      default: () => randomBytes(18).toString('hex'),
    },
    relationshipType: {
      type: String,
      enum: ['guardian', 'safeWalker'],
      required: true,
    },
    status: {
      type: String,
      enum: ['pending', 'accepted', 'expired', 'cancelled'],
      default: 'pending',
      required: true,
    },
    expiresAt: {
      type: Date,
      required: true,
      index: true,
    },
  },
  {
    timestamps: true,
  },
);

export type ConnectionInviteDocument = HydratedDocument<ConnectionInvite>;
export type ConnectionInviteModel = Model<ConnectionInvite>;

export const ConnectionInviteModel = model<ConnectionInvite>(
  'ConnectionInvite',
  connectionInviteSchema,
);

File: mobile/lib/models/connection_invite.dart
class ConnectionInvite {
  const ConnectionInvite({
    required this.id,
    required this.token,
    required this.relationshipType,
    required this.status,
    required this.expiresAt,
    required this.deepLink,
    required this.shareLink,
  });

  final String id;
  final String token;
  final String relationshipType;
  final String status;
  final DateTime expiresAt;
  final String deepLink;
  final String shareLink;

  factory ConnectionInvite.fromJson(Map<String, dynamic> json) {
    return ConnectionInvite(
      id: json['id'] as String,
      token: json['token'] as String,
      relationshipType: json['relationshipType'] as String,
      status: json['status'] as String,
      expiresAt: DateTime.parse(json['expiresAt'] as String),
      deepLink: json['deepLink'] as String,
      shareLink: json['shareLink'] as String,
    );
  }
}
